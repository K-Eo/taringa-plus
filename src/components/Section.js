import React, { Component } from 'react'
import { Menu } from 'semantic-ui-react'
import Posts from './Posts'
import './Section.css'

export class Section extends Component {
  componentDidMount () {
    const { category, count, id, trending } = this.props
    this.props.fetch(id, category, count, trending)
  }

  render () {
    const { error, name, status, posts } = this.props

    return (
      <div className='section'>
        <Menu pointing secondary className='section-nav'>
          <Menu.Item active>
            {name}
          </Menu.Item>
        </Menu>
        <Posts
          fetchControl={{ error: error, status: status }}
          posts={posts}
        />
      </div>
    )
  }
}

Section.defaultProps = {
  error: '',
  fetch: () => {},
  name: '#Section',
  posts: [],
  status: 'success'
}

export default Section
