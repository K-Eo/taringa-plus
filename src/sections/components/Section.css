.section-nav .active.item {
  font-size: 1.3rem;
  padding-left: 0 !important;
  padding-right: 0 !important;
}

.section ~ .section {
  margin-top: 40px;
}
