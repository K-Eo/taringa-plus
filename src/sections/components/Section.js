import React, { Component } from 'react'
import { Menu } from 'semantic-ui-react'
import Posts from '../../posts/components/Posts'
import './Section.css'

export class Section extends Component {
  componentDidMount () {
    const { category, count, id, trending } = this.props
    this.props.fetch(id, category, count, trending)
  }

  componentWillUnmount () {
    this.props.clear(this.props.id)
  }

  render () {
    const { error, name, status, posts } = this.props

    return (
      <div className='section'>
        <Menu pointing secondary className='section-nav'>
          <Menu.Item active>
            {name}
          </Menu.Item>
        </Menu>
        <Posts
          fetchControl={{ error: error, status: status }}
          posts={posts}
        />
      </div>
    )
  }
}

Section.defaultProps = {
  error: '',
  fetch: () => {},
  name: '#Section',
  posts: [],
  status: 'success',
  clear: () => {}
}

export default Section
