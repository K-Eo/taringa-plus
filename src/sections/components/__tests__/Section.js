import React from 'react'
import renderer from 'react-test-renderer'
import { Section } from '../Section'

describe('Section', () => {
  it('renders', () => {
    const component = renderer.create(
      <Section/>
    )

    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('loads post when componentDidMount', () => {
    const mock = jest.fn()

    renderer.create(
      <Section fetch={mock}/>
    )

    expect(mock).toBeCalled()
  })
})
