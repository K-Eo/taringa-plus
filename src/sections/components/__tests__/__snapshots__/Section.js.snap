// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section renders 1`] = `
<div
  className="section"
>
  <div
    className="ui pointing secondary section-nav menu"
  >
    <div
      className="active item"
      onClick={[Function]}
    >
      #Section
    </div>
  </div>
  <div
    className="ui icon message"
  >
    <i
      aria-hidden="true"
      className="smile icon"
    />
    <div
      className="content"
    >
      <div
        className="header"
      >
        No hay posts!
      </div>
      <p>
        Prueba recargando la página para intentar descargar los últimos posts.
      </p>
    </div>
  </div>
</div>
`;
